core/domain/learner_progress_services.py

```diff
@@ -0,0 +1,404 @@
+        completed_collection_ids=activities_completed.completed_collection_ids)
+
+    activities_completed_model.put()
+
+def save_incomplete_explorations(incomplete_explorations,
+                                 last_playthrough_information=None):
+    """Save an incomplete explorations domain object as an
+    IncompleteExplorationsModel entity in the datastore.
+
+    Args:
+        incomplete_explorations: IncompleteExplorations. The incomplete
+            explorations domain object to be saved in the datastore.
+    """
+    incomplete_explorations_model = user_models.IncompleteExplorationsModel(
+        id=incomplete_explorations.id,
+        incomplete_exploration_ids=(
+            incomplete_explorations.incomplete_exploration_ids))
+
+    if last_playthrough_information:
+        last_playthrough_information_model = (
+            user_models.ExpUserLastPlaythroughModel(
+                id=last_playthrough_information.id,
+                user_id=last_playthrough_information.user_id,
+                exploration_id=last_playthrough_information.exploration_id,
+                version_last_played=(
+                    last_playthrough_information.version_last_played),
+                time_last_played_msec=(
+                    last_playthrough_information.time_last_played_msec),
+                last_state_played=(
+                    last_playthrough_information.last_state_played)))
+        last_playthrough_information_model.put()
+
+    incomplete_explorations_model.put()
+
+def save_incomplete_collections(incomplete_collections):
+    """Save an incomplete collections domain object as an
+    IncompleteCollectionsModel entity in the datastore.
+
+    Args:
+        incomplete_collections: IncompleteCollections. The incomplete
+            collections domain object to be saved in the datastore.
+    """
+    incomplete_collections_model = user_models.IncompleteCollectionsModel(
+        id=incomplete_collections.id,
+        incomplete_collection_ids=(
+            incomplete_collections.incomplete_collection_ids))
+
+    incomplete_collections_model.put()
+
+def mark_exploration_as_completed(user_id, exp_id):
+    """Adds the exploration id to the completed list of the user unless the
+    exploration has already been completed or has been created by the user.
+    It is also removed from the incomplete list (if present).
+
+    Args:
+        user_id: str. The id of the user who has completed the exploration.
+        exp_id: str. The id of the completed exploration.
+    """
+    activities_completed_model = (
+        user_models.ActivitiesCompletedByLearnerModel.get(
+            user_id, strict=False))
+    if not activities_completed_model:
+        activities_completed_model = (
+            user_models.ActivitiesCompletedByLearnerModel(id=user_id))
+
+    exploration_created_ids = (
+        subscription_services.get_exploration_ids_subscribed_to(user_id))
```

Comment:
Changed the variable name. The objective over here is that we don't want anything that appears in the user's creator dashboard to appear in the learner dashboard. Since the subscribed explorations (edited/created) appear in the creator dashboard they will not appear in the learner dashboard.